### PR TITLE
Improve documentation of RecordNotSaved and RecordNotDestroyed [ci-skip]

### PR DESCRIPTION
**Pull request by p8**
Add examples and explain that RecordNotDestroyed triggered by throwing :abort in callbacks.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: activerecord/lib/active_record/errors.rb [hunk 1]
```diff
@@ -130,9 +130,17 @@ def initialize(message = nil, model = nil, primary_key = nil, id = nil)
 
   # Raised by {ActiveRecord::Base#save!}[rdoc-ref:Persistence#save!] and
   # {ActiveRecord::Base.update_attribute!}[rdoc-ref:Persistence#update_attribute!]
-  # methods when a record is failed to validate or cannot be saved due to any of the
+  # methods when a record failed to validate or cannot be saved due to any of the
   # <tt>before_*</tt> callbacks throwing +:abort+. See
   # ActiveRecord::Callbacks for further details
```
Comment (carlosantoniodasilva) [inline activerecord/lib/active_record/errors.rb:135]:
```suggestion
  # ActiveRecord::Callbacks for further details.
```

File: activerecord/lib/active_record/errors.rb [hunk 2]
```diff
@@ -143,15 +151,17 @@ def initialize(message = nil, record = nil)
   end
 
   # Raised by {ActiveRecord::Base#destroy!}[rdoc-ref:Persistence#destroy!]
-  # when a call to {#destroy}[rdoc-ref:Persistence#destroy]
-  # would return false.
+  # when a record cannot be destroyed due to any of the
+  # <tt>before_destroy</tt> callbacks throwing +:abort+. See
+  # ActiveRecord::Callbacks for further details
```
Comment (carlosantoniodasilva) [inline activerecord/lib/active_record/errors.rb:156]:
```suggestion
  # ActiveRecord::Callbacks for further details.
```

**Review by carlosantoniodasilva — COMMENTED** [review]
Just thought of adding a final period on a couple sentences, otherwise lgtm 👍

**Review by MatheusRich — APPROVED** [review]

**Comment by p8** [comment]
Thanks @carlosantoniodasilva and @MatheusRich !

**Pull request merged**